api/v1beta1/keystoneapi.go

```
@@ -212,3 +219,48 @@ func GetScopedAdminServiceClient(
 
 	return os, ctrl.Result{}, nil
 }
+
+// GetKeystoneOverrides - Get keystone override configuration from a secret by label.
+// Human operators can manually generate a secret to override most of the keystone
+// configuration to point to a central keystone instance.
+// Only a subset of supported keys present in the secret are extracted and returned,
+// any other information is currently ignored.
+func GetKeystoneOverrides(
+	ctx context.Context,
+	h *helper.Helper,
+	namespace string,
+	labelSelector map[string]string,
+) (map[string]string, error) {
+
+	// Process the overrides
+	overrides := map[string]string{}
+
+	secrets, err := secret.GetSecrets(ctx, h, namespace, labelSelector)
+	// Use direct label selector string for key existence check
+	if err != nil {
+		return overrides, fmt.Errorf("could not get secrets: %w", err)
+	}
+	// list is empty, return
+	if len(secrets.Items) == 0 {
+		h.GetLogger().Info("No keystone overrides, using default configuration")
+		return overrides, nil
+	}
+	// Invalid case: more than one secret with the same label is present in the
+	// same namespace. This is not allowed and an error must be returned
+	if len(secrets.Items) > 1 {
```

I wonder  if we should treat this case differently. Right now, we are not returning any error and when more than one secret with the label are present in the env, the glance reconciles to keystone defaults, no matter what was the previous state. So, if we had overrides already mounted, pods revert to defaults. Maybe in this case we should log the info (as we do now), but not rollout new pods to prevent possible disruption or unnecessary config change?

---

Thank you, this is an interesting perspective. It makes sense to me and we can return an error, with the right message, to prevent Glance (or any other service) to reconcile, until the secret is fixed.
It shouldn't happen to see two secrets with the same `LabelSelector`, but I agree that we should preserve the service configuration in this case.